### PR TITLE
create calculator structure to resolve sum operation

### PR DESCRIPTION
**Pull request by matiaslev**

### CODE REVIEW

**Review by gaston-mira-glb — APPROVED** [review]

File: app/src/main/java/com/globant/counter/android/mvp/model/CalculatorModel.java [hunk 1]
```diff
@@ -0,0 +1,14 @@
+package com.globant.counter.android.mvp.model;
+
+public class CalculatorModel {
+
+    public int calculate(int firstNumber, String operationType, int secondNumber) throws Exception {
+        OperationInterface operation = new SumOperationClass();
```
Comment (nacho-globant) [inline app/src/main/java/com/globant/counter/android/mvp/model/CalculatorModel.java]: I think that you dont need create a hierarchy to make a simple operation.

File: app/src/main/java/com/globant/counter/android/mvp/presenter/CalculatorPresenter.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+import android.app.Activity;
+
+import com.globant.counter.android.mvp.model.CalculatorModel;
+import com.globant.counter.android.mvp.view.CalculatorView;
+import com.globant.counter.android.util.bus.RxBus;
+import com.globant.counter.android.util.bus.observers.EqualsButtonPressedObserver;
+
+public class CalculatorPresenter {
+
+    private CalculatorModel model;
```
Comment (nacho-globant) [inline app/src/main/java/com/globant/counter/android/mvp/presenter/CalculatorPresenter.java]: mark as final

File: app/src/main/java/com/globant/counter/android/mvp/presenter/CalculatorPresenter.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    private CalculatorView view;
```
Comment (nacho-globant) [inline app/src/main/java/com/globant/counter/android/mvp/presenter/CalculatorPresenter.java]: mark as final

File: app/src/main/java/com/globant/counter/android/mvp/view/CalculatorView.java [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+        ButterKnife.bind(this, activity);
+    }
+
+    public void showResult(String result) {
+        Toast.makeText(getContext(), result, Toast.LENGTH_LONG).show();
+    }
+
+    @OnClick(R.id.button_equals)
+    public void equalsButtonPressed() {
+        RxBus.post(new EqualsButtonPressedObserver.EqualsButton(
```
Comment (nacho-globant) [inline app/src/main/java/com/globant/counter/android/mvp/view/CalculatorView.java:32]: This is fine but to simplify the process you can specify the Onclick into the Activity and access to presenter directly

**Review by nacho-globant — COMMENTED** [review]
some comments

**Review by nacho-globant — APPROVED** [review]
+1

**Pull request merged**